core/src/main/java/org/infinispan/container/impl/InternalEntryFactoryImpl.java

```diff
@@ -185,11 +185,11 @@ public CacheEntry copy(CacheEntry cacheEntry) {
    }
 
    @Override
-   public InternalCacheValue getValueFromCtxOrCreateNew(Object key, InvocationContext ctx) {
+   public <K, V> InternalCacheValue<V> getValueFromCtxOrCreateNew(K key, InvocationContext ctx) {
```

Comment:
-1 for the generic arguments, the callers ignore them anyway.
Also -1 for `CreateNew`, to me that implies that the result will never be `null`.

And finally, I'm not sure why we need this method, because all the callers immediately use it to create a `MarshalledEntry`. @ryanemerson do you remember why this was needed?

Reply:
So originally this method was just in `CacheWriterInterceptor` as `getStoredValue` (seemingly a much better name), but the same logic was required in `TxBatchUpdater` which is why I added this method to `InternalEntryFactory`. 

I think this method is required so that the correct version is retrieved for the metadata. This metadata is then persisted as part of the `MarshalledEntry`.

Reply:
There are generics on all the other methods, I was making it consistent with them.

Also I agree re: `CreateNew` I just didn't have time last night to remove it.

Reply:
Oh and also re: generics they are already on the interface, so there is no reason not to put them here really.

https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/container/impl/InternalEntryFactory.java#L168

Reply:
Oh well, at least only the methods are parametrized, not the interface itself :))